src/Service/NotificationReceiverService.php

```diff
@@ -0,0 +1,378 @@
+                array(
+                    'success' => false,
+                    'message' => $message
+                )
+            );
+        }
+    }
+
+    /**
+     * Validation of the notification
+     * Testing the notification against the plugin environment and the HMAC signature
+     *
+     * @param $notification
+     * @param $merchantAccount
+     * @param $hmacKey
+     * @return bool
+     * @throws AdyenException
+     * @throws HMACKeyValidationException
+     * @throws MerchantAccountCodeException
+     */
+    protected function isValidated($notification, $merchantAccount, $hmacKey)
+    {
+        // Check if the notification is a test notification
+        $isTestNotification = $this->isTestNotificationPspReference($notification['pspReference']);
+
+        // Validate if notification or configuration merchant account value is missing
+        if (empty($notification['merchantAccountCode']) || empty($merchantAccount)) {
+            if ($isTestNotification) {
+                $message = 'MerchantAccountCode or merchant account configuration is empty.';
+                throw new MerchantAccountCodeException($message);
+            }
+
+            return false;
+        }
+
+        // Validate HMAC signature
+        if (!$this->hmacSignature->isValidNotificationHMAC($hmacKey, $notification)) {
+            if ($isTestNotification) {
+                $message = 'HMAC key validation failed';
+                throw new HMACKeyValidationException($message);
+            }
+
+            return false;
+        }
+
+        // Notification is validated
+        return true;
+    }
+
+    /**
+     * Authorize notification based on Basic Authentication
+     *
+     * @param $isTestNotification
+     * @param $requestUser
+     * @param $requestPassword
+     * @return bool
+     * @throws AuthenticationException
+     */
+    private function isAuthorized($isTestNotification, $requestUser, $requestPassword)
+    {
+        // Retrieve username and password from config
+        $userName = $this->configurationService->getNotificationUsername();
+        $password = $this->configurationService->getNotificationPassword();
+
+        // Validate if username and password is sent
+        if ((is_null($requestUser) || is_null($requestPassword))) {
+            if ($isTestNotification) {
+                $message = 'PHP_AUTH_USER or PHP_AUTH_PW is not in the request.';
+                throw new AuthenticationException($message);
+            }
+
+            return false;
+        }
+
+        // Validate if username and password is configured
+        if ((is_null($userName) || is_null($password))) {
+            if ($isTestNotification) {
+                $message = 'Notifications username and/or password are not configured.';
+                throw new AuthenticationException($message);
+            }
+
+            return false;
+        }
+
+        // Validate the username and password
```

Comment:
don't we already have these in the PHP api lib? Create ticket to replace if yes

Reply:
PW-2843